src/core/exec-invoke.c

```diff
@@ -4995,7 +4995,7 @@ int exec_invoke(
 #if ENABLE_SMACK
                 /* LSM Smack needs the capability CAP_MAC_ADMIN to change the current execution security context of the
                  * process. This is the latest place before dropping capabilities. Other MAC context are set later. */
-                if (use_smack && context->smack_process_label) {
+                if (use_smack && (params->fallback_smack_process_label || context->smack_process_label)) {
```

Comment:
I'd prefer that we drop the conditionalization here. setup_smack is a no-op when neither is specified anyways.